src/modules/battery-upower.c

```diff
@@ -164,6 +172,11 @@ upowbat_update(void)
 		}
 	}
 
+	if (g_strcmp0(upowbat.supply_level, supply_level) != 0) {
+		mce_log(LL_DEBUG, "%s: Capacity Level: %s -> %s", MODULE_NAME, upowbat.supply_level, supply_level);
+		upowbat.supply_level = supply_level;
```

Comment:
g_free(upowbat.supply_level);
upowbat.supply_level = supply_level;

Reply:
thank you!